File: migration-backend/pkg/model/likecoin_migration.go
package model

import (
	"time"
)

type LikeCoinMigrationStatus string

const (
	LikeCoinMigrationStatusPendingCosmosTxHash LikeCoinMigrationStatus = "pending_cosmos_tx_hash"
	LikeCoinMigrationStatusVerifyingCosmosTx   LikeCoinMigrationStatus = "verifying_cosmos_tx"
	LikeCoinMigrationStatusEvmMinting          LikeCoinMigrationStatus = "evm_minting"
	LikeCoinMigrationStatusEvmVerifying        LikeCoinMigrationStatus = "evm_verifying"
	LikeCoinMigrationStatusCompleted           LikeCoinMigrationStatus = "completed"
	LikeCoinMigrationStatusFailed              LikeCoinMigrationStatus = "failed"
)

func (s *LikeCoinMigrationStatus) IsValid() bool {
	switch *s {
	case LikeCoinMigrationStatusPendingCosmosTxHash,
		LikeCoinMigrationStatusVerifyingCosmosTx,
		LikeCoinMigrationStatusEvmMinting,
		LikeCoinMigrationStatusEvmVerifying,
		LikeCoinMigrationStatusCompleted,
		LikeCoinMigrationStatusFailed:
		return true
	}
	return false
}

type LikeCoinMigration struct {
	Id                   uint64
	CreatedAt            time.Time
	UserCosmosAddress    string
	BurningCosmosAddress string
	MintingEthAddress    string
	UserEthAddress       string
	EvmSignature         string
	EvmSignatureMessage  string
	Amount               string

	Status       LikeCoinMigrationStatus
	CosmosTxHash *string
	EvmTxHash    *string
	FailedReason *string
}

File: migration-admin/apis/models/likecoinMigration.ts
import { z } from "zod";

export enum LikeCoinMigrationStatus {
  PendingCosmosTxHash = "pending_cosmos_tx_hash",
  VerifyingCosmosTx = "verifying_cosmos_tx",
  EvmMinting = "evm_minting",
  EvmVerifying = "evm_verifying",
  Completed = "completed",
  Failed = "failed",
}

export const LikeCoinMigrationSchema = z.object({
  id: z.number(),
  created_at: z.coerce.date(),
  user_cosmos_address: z.string(),
  user_eth_address: z.string(),
  evm_signature: z.string(),
  amount: z.string(),
  status: z.nativeEnum(LikeCoinMigrationStatus),
  cosmos_tx_hash: z.string().nullable(),
  evm_tx_hash: z.string().nullable(),
  failed_reason: z.string().nullable(),
});

export type LikeCoinMigration = z.infer<typeof LikeCoinMigrationSchema>;
